### PR TITLE
[HVAC] Clear the active preset when the relevant setpoint attributes are set manually

### PR DESCRIPTION
**Pull request by hasty**
Adds functionality to clear the ActivePresetHandle attribute when either of the active OUCH attributes get changed.

This should not be merged until this spec change is made: https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/10336

### CODE REVIEW

File: examples/thermostat/thermostat-common/include/thermostat-delegate-impl.h [hunk 1]
```diff
@@ -58,7 +58,7 @@ class ThermostatDelegate : public Delegate
 
     void InitializePendingPresets() override;
 
-    CHIP_ERROR AppendToPendingPresetList(const Structs::PresetStruct::Type & preset) override;
+    CHIP_ERROR AppendToPendingPresetList(const PresetStructWithOwnedMembers & preset) override;
```
Comment (bzbarsky-apple) [inline examples/thermostat/thermostat-common/include/thermostat-delegate-impl.h:61]: This looks like bits of https://github.com/project-chip/connectedhomeip/pull/35161.  Needs rebasing to tip?

Comment (hasty) [reply examples/thermostat/thermostat-common/include/thermostat-delegate-impl.h]: Yeah, #35161 just got merged, so I hit update and the diff is gone.

File: src/app/clusters/thermostat-server/thermostat-server.cpp [hunk 2]
```diff
@@ -468,6 +468,52 @@ void ThermostatAttrAccess::OnFabricRemoved(const FabricTable & fabricTable, Fabr
...
+void MatterThermostatClusterServerAttributeChangedCallback(const chip::app::ConcreteAttributePath & attributePath)
+{
+    uint32_t flags;
+    if (FeatureMap::Get(attributePath.mEndpointId, &flags) != Status::Success)
+    {
+        ChipLogError(Zcl, "MatterThermostatClusterServerAttributeChangedCallback: could not get feature flags");
+        return;
+    }
+
+    auto featureMap = chip::BitMask<Feature, uint32_t>(flags);
```
Comment (bzbarsky-apple) [inline src/app/clusters/thermostat-server/thermostat-server.cpp:480]:
```suggestion
    auto featureMap = BitMask<Feature, uint32_t>(flags);
```

File: src/app/clusters/thermostat-server/thermostat-server.cpp [hunk 3]
```diff
@@ -468,6 +468,52 @@ void ThermostatAttrAccess::OnFabricRemoved(const FabricTable & fabricTable, Fabr
...
+    if (!featureMap.Has(Feature::kPresets))
+    {
+        // This server does not support presets, so nothing to do
+        return;
+    }
+
+    bool occupied = true;
+    if (featureMap.Has(Feature::kOccupancy))
+    {
+        chip::BitMask<OccupancyBitmap, uint8_t> occupancy;
```
Comment (bzbarsky-apple) [inline src/app/clusters/thermostat-server/thermostat-server.cpp:490]:
```suggestion
        BitMask<OccupancyBitmap, uint8_t> occupancy;
```

File: src/app/clusters/thermostat-server/thermostat-server.cpp [hunk 4]
```diff
@@ -762,6 +808,11 @@ MatterThermostatClusterServerPreAttributeChangedCallback(const app::ConcreteAttr
     }
 }
 
+void MatterThermostatClusterServerAttributeChangedCallback(const chip::app::ConcreteAttributePath & attributePath)
```
Comment (bzbarsky-apple) [inline src/app/clusters/thermostat-server/thermostat-server.cpp:811]:
```suggestion
void MatterThermostatClusterServerAttributeChangedCallback(const ConcreteAttributePath & attributePath)
```

File: src/app/clusters/thermostat-server/thermostat-server.cpp [hunk 5]
```diff
@@ -468,6 +468,52 @@ void ThermostatAttrAccess::OnFabricRemoved(const FabricTable & fabricTable, Fabr
...
```
Comment (bzbarsky-apple) [inline src/app/clusters/thermostat-server/thermostat-server.cpp:471]:
```suggestion
void MatterThermostatClusterServerAttributeChangedCallback(const ConcreteAttributePath & attributePath)
```

**Review by bzbarsky-apple — APPROVED** [review]

**Review by bauerschwan — APPROVED** [review]

File: src/app/clusters/thermostat-server/thermostat-server.cpp [hunk 6]
```diff
@@ -468,6 +468,52 @@ void ThermostatAttrAccess::OnFabricRemoved(const FabricTable & fabricTable, Fabr
...
+        break;
+    case UnoccupiedHeatingSetpoint::Id:
+    case UnoccupiedCoolingSetpoint::Id:
+        clearActivePreset = !occupied;
+        break;
+    }
+    if (!clearActivePreset)
+    {
+        return;
+    }
```
[the remark this answers is not in the document]
Comment (bauerschwan) [reply src/app/clusters/thermostat-server/thermostat-server.cpp:512]: Just cosmetic, so I resolve

**Pull request merged**